### PR TITLE
Run PSC Controller when enabled

### PR DESCRIPTION
**Pull request by swetharepakula**
This PR is dependent on #1381

### CODE REVIEW

File: pkg/psc/controller.go [hunk 1]
```diff
@@ -75,16 +79,96 @@ type Controller struct {
...
+			case <-stopChan:
+				return
+			default:
+				processKey()
+			}
+		}
+	}, time.Second, stopChan)
+
+	go func() {
+		time.Sleep(2 * time.Minute)
```
Comment (freehan) [inline pkg/psc/controller.go:139]: make 2 mins a constant?

File: pkg/psc/controller.go [hunk 2]
```diff
@@ -75,16 +79,96 @@ type Controller struct {
...
+		return c.hasSynced(), nil
+	}, stopChan)
+
+	klog.V(2).Infof("Starting private service connect controller")
+	defer func() {
+		klog.V(2).Infof("Shutting down private service connect controller")
+		c.svcAttachmentQueue.ShutDown()
+	}()
+
+	go wait.Until(func() {
```
Comment (freehan) [inline pkg/psc/controller.go:117]: maybe worth adding this into a separate `sync` function?

and then simply do go wait.Util(c.sync(), time.Second, stopChan)?

File: pkg/psc/controller.go [hunk 3]
```diff
@@ -75,16 +79,96 @@ type Controller struct {
...
 		saClient:            ctx.SAClient,
 		saNamer:             saNamer,
 		svcAttachmentLister: ctx.SAInformer.GetIndexer(),
+		svcAttachmentQueue:  workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
 		serviceLister:       ctx.ServiceInformer.GetIndexer(),
 		hasSynced:           ctx.HasSynced,
 		recorder:            ctx.Recorder,
 	}
+
+	ctx.SAInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```
Comment (freehan) [inline pkg/psc/controller.go:94]: no need for DeleteFunc because of finalizer?

**Review by freehan — COMMENTED** [review]
just nits. mostly LGTM

File: pkg/psc/controller.go (hunk 3 above)
Comment (swetharepakula) [reply pkg/psc/controller.go:97]: Yes, that's correct. We don't do anything on delete, so not processing on those. Deletion will eventually be handled by GC

**Comment by swetharepakula** [comment]
Since #1381 is merged and this pr has been rebased, removing hold.

/hold cancel

**Pull request merged**